lib/ecto/adapters/postgres.ex

```diff
@@ -79,66 +81,40 @@ defmodule Ecto.Adapters.Postgres do
       extra = extra <> " LC_CTYPE='#{lc_ctype}'"
     end
 
-    {output, status} =
-      run_with_psql opts, "CREATE DATABASE \"" <> database <>
-                          "\" ENCODING='#{encoding}'" <> extra
+    command = "CREATE DATABASE #{database} " <>
+              "ENCODING '#{encoding}'" <> extra
 
-    cond do
-      status == 0                       -> :ok
-      String.contains?(output, "42P04") -> {:error, :already_up}
-      true                              -> {:error, output}
+    case run_query(opts, command) do
+      {:ok, _result} ->
+        :ok
+      {:error, %Postgrex.Error{postgres: %{code: :duplicate_database}}} ->
+        {:error, :already_up}
+      error -> error
     end
   end
 
   @doc false
   def storage_down(opts) do
-    {output, status} = run_with_psql(opts, "DROP DATABASE \"#{opts[:database]}\"")
-
-    cond do
-      status == 0                       -> :ok
-      String.contains?(output, "3D000") -> {:error, :already_down}
-      true                              -> {:error, output}
+    command = "DROP DATABASE #{opts[:database]}"
+
+    case run_query(opts, command) do
+      {:ok, _result} ->
+        :ok
+      {:error, %Postgrex.Error{postgres: %{code: :invalid_catalog_name}}} ->
+        {:error, :already_down}
+      error -> error
```

Comment:
Just a heads up, this is not backwards compatible. This used to return `{:error, String}` and is now returning `{:error, Postgrex.Error}`